exceptiongroup/_tools.py

```diff
@@ -6,6 +6,25 @@
 from . import ExceptionGroup
 
 
+def leaf_exceptions(exc):
+    """ Iterate and extract sub exceptions from given exception.
```

Comment:
Can you expand upon this a little bit? I don't think the current docstring would be clear to someone who's not already super familiar with ExceptionGroup. Maybe some examples would help clarify?